src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -1303,6 +1303,48 @@ fn generic_simd_intrinsic(
         return Ok(bx.select(m_i1s, args[1].immediate(), args[2].immediate()));
     }
 
+    if name == "simd_bitmask" {
+        // The `fn simd_bitmask(vector) -> unsigned integer` intrinsic takes a
+        // vector mask and returns an unsigned integer containing the first bit
+        // of each lane.
+
+        // If the vector has less than 8 lanes, an u8 is returned with zeroed
+        // trailing bits.
+        let expected_int_bits = in_len.max(8);
+        match ret_ty.sty {
+            // Note: it's intentional that these `.unwrap()` crash for
+            // isize/usize, since we need the number of lanes to match exactly
+            // with the number of bits:
+            ty::Uint(i) if expected_int_bits == i.bit_width().unwrap() => (),
+            _ => return_error!(
+                "bitmask `{}`, expected `u{}`",
+                ret_ty, expected_int_bits
+            ),
+        }
+
+        // Integer vector <i_ x in_len>:
+        let i_xn = match in_elem.sty {
+            ty::Int(_) | ty::Uint(_) => args[0].immediate(),
+            ty::Float(i) => {
```

Comment:
Honestly, I didn't gave this much thought and I don't have strong feelings about this. 

This intrinsic has to be generic to support vectors of different lengths, so I just added support for creating bitmasks for all `repr(simd)` vectors that we currently support.

We could just support unsigned integer vectors, letting users do the bitcasts to them from signed integer and floating-point vectors, but I don't see a huge win on doing that either.

Reply:
Now that this does what it was supposed to do (extract the MSB), it is a bit more useful than before for floating-point vectors, but AFAICT user code can still just do a transmute. Should I remove floating-point support ? `packed_simd` does not have "floating-point" masks, so support for that will probably not be used, although it might be helpful for playing with codegen of floating-point masks in godbolt and such.

Reply:
Yeah now it's more useful, but if it's not going to be exercised normally and it has an extra (if small) code path I would prefer to err on the side of not supporting it.

Reply:
Will fix in next commit!